src/gnutls_crypto.cpp

```diff
@@ -342,6 +347,18 @@ namespace oxen::quic
         set_tls_hook_functions();
     }
 
+    std::string_view GNUTLSSession::selected_alpn()
+    {
+        gnutls_datum_t proto;
+        if (auto rv = gnutls_alpn_get_selected_protocol(session, &proto); rv < 0)
+        {
+            log::warning(log_cat, "{} called, but ALPN negotiation incomplete.", __PRETTY_FUNCTION__);
+            return ""sv;
+        }
+
+        return proto.size ? std::string_view{(const char*)proto.data, proto.size} : ""sv;
```

Comment:
Maybe return an optional instead? Might make the return logic and return verification in the calling scope simpler

This is making me wonder if it's worth putting in the time to wrap `gnutls_datam_t` in a lightweight wrapper similar to the artist formerly known as `datum` (currently `x509_loader`)